numpy/core/src/umath/ufunc_object.c

```diff
@@ -1880,12 +1879,15 @@ _parse_axes_arg(PyUFuncObject *ufunc, PyObject *axes, PyArrayObject **op,
                 j++;
             }
         }
-        do_remap[i] = (j < op_ndim);
+        if (j == op_ndim) {
+            remap_axis[i] = NULL;
+        }
     } /* end of for(i) loop over operands */
     return 0;
 }
 
-#define REMAP_AXIS(i, j) ((axes != NULL && do_remap[i]) ? remap_axis[i][j] : j)
+#define REMAP_AXIS(i, j) ((remap_axis != NULL && \
```

Comment:
Can you rename these to `op, axis`, instead of `i, j`?

Reply:
Yes, better. Done.
...  (though I took your initial `op, axis` from the e-mail version I received; can change if you prefer)